libcloud/storage/drivers/s3.py

```diff
@@ -419,7 +419,7 @@ def download_object_as_stream(self, obj, chunk_size=None):
         return self._get_object(
             obj=obj, callback=read_in_chunks,
             response=response,
-            callback_kwargs={'iterator': response.iter_content,
+            callback_kwargs={'iterator': response.iter_content(),
```

Comment:
@pquentin I'm also curious to see a performance difference between the 2. The old code buffered out the download to get the stream size and then returned the iterator. I think for a large file that would really hurt both time and memory consumption